### PR TITLE
Workin Joey Talks

### PR DESCRIPTION
**Pull request by Ben10083**
# About the pull request

Expansion on #4042 that was discussed with Thwomp. Working Joe randomly says a quip and responds when attacked.

# Explain why it's good for the game

More character to Joey instead of being a static prop that doesn't move. Showing how "Joey" is still active but been interfered with by the crew. 

# Testing Photographs and Procedure

image

# Changelog
:cl:
add: Workin Joey now randomly talks and responds when attacked
/:cl:

### CODE REVIEW

**Comment by Ben10083** [comment]
I am sorry for the attack code. Me and @harryob tried to find a better way but signals just don't work...

File: code/game/objects/structures/props.dm [hunk 1]
```diff
@@ -1158,3 +1158,76 @@
...
+	unslashable = FALSE
+	wrenchable = FALSE
+	/// converted into minutes when used to determine cooldown timer between quips
+	var/quip_delay_minimum = 5
+	/// delay between Quips. Slightly randomized with quip_delay_minimum plus a random number
+	COOLDOWN_DECLARE(quip_delay)
+	/// delay between attack voicelines. Short but done for anti-spam
+	COOLDOWN_DECLARE(damage_delay)
+	/// list of quip emotes, taken from Working Joe
+	var/list/quips = list(
```
Comment (Zonespace27) [inline code/game/objects/structures/props.dm:1170]: static list

File: code/game/objects/structures/props.dm [hunk 2]
```diff
@@ -1158,3 +1158,76 @@
...
+		/datum/emote/living/carbon/human/synthetic/working_joe/quip/alwaysknow_damaged,
+		/datum/emote/living/carbon/human/synthetic/working_joe/quip/not_liking,
+		/datum/emote/living/carbon/human/synthetic/working_joe/greeting/how_can_i_help,
+		/datum/emote/living/carbon/human/synthetic/working_joe/task_update/day_never_done,
+		/datum/emote/living/carbon/human/synthetic/working_joe/task_update/required_by_apollo,
+		/datum/emote/living/carbon/human/synthetic/working_joe/warning/safety_breach
+	)
+	/// list of voicelines to use when damaged
+	var/list/damaged = list(
```
Comment (Zonespace27) [inline code/game/objects/structures/props.dm:1179]: static list

File: code/game/objects/structures/props.dm [hunk 3]
```diff
@@ -1158,3 +1158,76 @@
...
+		emote(pick(damaged))
+		COOLDOWN_START(src, damage_delay, 8 SECONDS)
+
+/// SAY THE LINE JOE
+/obj/structure/prop/invuln/joey/proc/emote(datum/emote/living/carbon/human/synthetic/working_joe/emote)
+	if (!emote)
+		return FALSE
+
+	for(var/mob/mob in hearers(src, null))
+		mob.show_message("<span class='game say'><span class='name'>[src]</span> says, \"[initial(emote.say_message)]\"</span>", SHOW_MESSAGE_AUDIBLE)
```
Comment (Zonespace27) [inline code/game/objects/structures/props.dm:1220]: i hate that this is the best way to do it

File: code/game/objects/structures/props.dm [hunk 4]
```diff
@@ -1158,3 +1158,76 @@
...
+		COOLDOWN_START(src, quip_delay, delay MINUTES)
+
+// Advert your eyes.
+/obj/structure/prop/invuln/joey/attackby(obj/item/W, mob/user)
+	attacked()
+	. = ..()
+
+/obj/structure/prop/invuln/joey/bullet_act(obj/item/projectile/P)
+	attacked()
+	. = ..()
```
Comment (Zonespace27) [inline code/game/objects/structures/props.dm:1206]:
```suggestion
	return ..()
```

File: code/game/objects/structures/props.dm [hunk 5]
```diff
@@ -1158,3 +1158,76 @@
...
+	//check if quip_delay cooldown finished. If so, random chance it says a line
+	if(COOLDOWN_FINISHED(src, quip_delay) && prob(10))
+		emote(pick(quips))
+		var/delay = rand(3) + quip_delay_minimum
+		COOLDOWN_START(src, quip_delay, delay MINUTES)
+
+// Advert your eyes.
+/obj/structure/prop/invuln/joey/attackby(obj/item/W, mob/user)
+	attacked()
+	. = ..()
```
Comment (Zonespace27) [inline code/game/objects/structures/props.dm:1202]:
```suggestion
	return ..()
```

File: code/game/objects/structures/props.dm [hunk 6]
```diff
@@ -1158,3 +1158,76 @@
...
+		/datum/emote/living/carbon/human/synthetic/working_joe/warning/that_stings,
+		/datum/emote/living/carbon/human/synthetic/working_joe/warning/irresponsible,
+		/datum/emote/living/carbon/human/synthetic/working_joe/warning/this_is_futile,
+		/datum/emote/living/carbon/human/synthetic/working_joe/warning/hysterical,
+		/datum/emote/living/carbon/human/synthetic/working_joe/warning/patience
+	)
+
+/obj/structure/prop/invuln/joey/Initialize()
+	. = ..()
+	START_PROCESSING(SSobj, src)
```
Comment (Zonespace27) [inline code/game/objects/structures/props.dm:1190]: stop processing on Destroy()

**Review by Zonespace27 — CHANGES_REQUESTED** [review]

File: code/game/objects/structures/props.dm (hunk 3 above)
Comment (Ben10083) [reply code/game/objects/structures/props.dm:1220]: Same. 😭

File: code/modules/mob/living/carbon/xenomorph/attack_alien.dm [hunk 7]
```diff
@@ -982,3 +982,12 @@
 	var/matrix/A = matrix()
 	apply_transform(A)
 	stat &= ~BROKEN //Remove broken. MAGICAL REPAIRS
+
+//Misc
+/obj/structure/prop/invuln/joey/attack_alien(mob/living/carbon/xenomorph/Alien)
```
Comment (harryob) [inline code/modules/mob/living/carbon/xenomorph/attack_alien.dm:987]: `Alien` should be snakecased

File: code/game/objects/structures/props.dm [hunk 8]
```diff
@@ -1158,3 +1158,80 @@
...
+		mob.show_message("<span class='game say'><span class='name'>[src]</span> says, \"[initial(emote.say_message)]\"</span>", SHOW_MESSAGE_AUDIBLE)
+
+	var/list/viewers = get_mobs_in_view(7, src)
+	for(var/mob/current_mob in viewers)
+		if(!(current_mob.client?.prefs.toggles_langchat & LANGCHAT_SEE_EMOTES))
+			viewers -= current_mob
+	langchat_speech(initial(emote.say_message), viewers, GLOB.all_languages, skip_language_check = TRUE)
+
+	if(initial(emote.sound))
+		playsound(src.loc, initial(emote.sound), 50, FALSE)
```
Comment (harryob) [inline code/game/objects/structures/props.dm:1233]:
```suggestion
		playsound(loc, initial(emote.sound), 50, FALSE)
```

File: code/game/objects/structures/props.dm [hunk 9]
```diff
@@ -1158,3 +1158,80 @@
...
+		/datum/emote/living/carbon/human/synthetic/working_joe/warning/patience
+	)
+
+/obj/structure/prop/invuln/joey/Initialize()
+	. = ..()
+	START_PROCESSING(SSobj, src)
+
+/obj/structure/prop/invuln/joey/Destroy()
+	STOP_PROCESSING(SSobj, src)
+	. = ..()
```
Comment (harryob) [inline code/game/objects/structures/props.dm:1194]:
```suggestion
	return ..()
```

**Review by harryob — CHANGES_REQUESTED** [review]

**Review by harryob — APPROVED** [review]

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

**Pull request merged**